tsfile/src/main/java/org/apache/iotdb/tsfile/read/reader/series/FileSeriesReader.java

```diff
@@ -54,23 +55,13 @@ public FileSeriesReader(IChunkLoader chunkLoader, List<ChunkMetaData> chunkMetaD
    * @return True if current chunk has next batch data
    */
   public boolean hasNextBatch() throws IOException {
-
-    // current chunk has additional batch
     if (chunkReader != null && chunkReader.hasNextBatch()) {
       return true;
     }
 
-    // current chunk does not have additional batch, init new chunk reader
-    while (chunkToRead < chunkMetaDataList.size()) {
-
-      ChunkMetaData chunkMetaData = nextChunkMeta();
-      if (chunkSatisfied(chunkMetaData)) {
-        // chunk metadata satisfy the condition
-        initChunkReader(chunkMetaData);
-
-        if (chunkReader.hasNextBatch()) {
-          return true;
-        }
+    if (hasNextChunk()) {
+      if (chunkReader.hasNextBatch()) {
+        return true;
```

Comment:
I think here you should use while(hasNextChunk()) instead of if(hasNextChunk).
Because, even if the current chunkReader doesn't have the needed batch, it doesn't mean that next Chunk won't have.